lib/Infer/AliveDriver.cpp

```diff
@@ -783,7 +791,66 @@ IR::Type &souper::AliveDriver::getType(int n) {
   }
   return *TypeCache[n];
 }
+namespace souper {
+struct Problem {
```

Comment:
problem is a very vague name for a class, please try to give it something more descriptive

also, add a comment somewhere around here documenting the solution that is being implemented